### PR TITLE
PLU-115: [SST-4] More refactors and unit tests

### PR DESCRIPTION
**Pull request by pregnantboy**
### TL;DR

Refactored test execution steps retrieval and improved test run logic.

### What changed?

- Extracted `getTestExecutionSteps` logic into a separate helper function.
- Updated `testRun` function to remove duplicate lines of code.

### What to test
- [x] Test steps will automatically reflect last test execution data.
- [x] Test execution data should correspond to the variables dropdown in subsequent steps
- [x] Testing earlier steps should not remove test execution data from later steps that have been previously tested
- [x] If an earlier step fails during testing, it should show the error in the current step. However, this is only temporary and toggle the expansion of the accordion will remove the error. You could see the error message on the corresponding step.

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#724** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/724?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#708** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/708?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#707** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/707?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#706** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/706?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#680** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/680?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#679** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/679?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#678** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/678?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by m0nggh — APPROVED** [review]
tested that it works, lgtm

**Comment by pregnantboy** [comment]
### Merge activity

* **Aug 27, 6:30 AM EDT**: @pregnantboy started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/706).
* **Aug 27, 6:36 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/706) couldn't merge this pull request because a downstack PR #680 failed to merge.
